website/project/views/comment.py

```diff
@@ -15,104 +16,203 @@
 from website.notifications.emails import notify
 from website.filters import gravatar
 from website.models import Guid, Comment
+from website.addons.base import GuidFile
 from website.project.decorators import must_be_contributor_or_public
 from datetime import datetime
 from website.project.model import has_anonymous_link
+from website.project.views.node import _view_project, n_unread_comments
 
 
-def resolve_target(node, guid):
+@must_be_contributor_or_public
+def view_comments_project(auth, **kwargs):
+    """
+    Returns information needed to get comments for the total discussion page
+    """
+
+    node = kwargs['node'] or kwargs['project']
+    page = request.args.get('page', None)
+
+    OVERVIEW = "overview"
+
+    if not page:
+        ret = _view_comments_total()
+    elif page == OVERVIEW:
```

Comment:
This could use refactoring. There is a lot of conditional logic and duplicated code between the `_view_comments_*` functions. 

One possible refactor would be to use pubsub/signals:

```python
def view_comments_project(auth, **kwargs):
    # ...
     # send signal
    comments_viewed.send(..., page ...)
     #...

# subscriber updates user's comments timestamps as appropriate
@comments_viewed.connect
def on_comments_viewed(... page ...):
    _update_comments_viewed_timespamp(...page..)
```